### DISCUSSION TITLE
Mermaid Integration with NetworkX to enable topological information about the graphs

### DISCUSSION THREAD

**Discussion by PurviChaurasia** [mermaid-js/mermaid#5190] [Q&A]
Hi I was wondering if there is a way to extract the nodes and edges from a given graph mermaid script. I have been trying to use regular expression to do this, but have been unable to set a standard expression for this task. Is there an inbuilt function for this purpose?
On a side note, if there is a direct integration/workflow present as to how to convert a mermaid script to a networkX graph it would extremely helpful to get all necessary topological information about it. Please let me know as soon as possible. Thanks! @jgreywolf @aloisklink

**Comment by InonS** [comment]
Hi, @PurviChaurasia !

As it turns out (Googled it), there's https://github.com/MikeBergmann/graphml2mermaid/commit/34684740bc5361fcd98c4bccb33891765773ec08 from @MikeBergmann. See his blog post https://blog.mdb977.de/rendering-networkx-graphs-or-graphml-files-via-mermaid/

**Comment by aloisklink** [comment]
Sorry for the late reply @PurviChaurasia!

As far as I'm aware, there's no official way to extract the nodes and edges from a given graph mermaid code. However, you can use some of Mermaid's internal functions for this. **Just beware that it is very likely your code will break in future Mermaid releases if you use internal functions!**

There's a `mermaid-to-excalidraw` package available under the MIT license that does something similar, so you could probably adapt it to use networkX or whichever format you want, see https://github.com/excalidraw/mermaid-to-excalidraw/blob/2a20a2e7bfedf68d184a435dff9ab67c43e1fc90/src/parser/flowchart.ts
